### PR TITLE
#146143817 Configure heroku deployment

### PR DESCRIPTION
**Pull request by collinmutembei**
#### What does this PR do?
Adds configuration files needed for the project to be deployed on heroku.

#### Description of Task to be completed?
- [x] Use the web interface to create heroku apps:
  - [x] [`hc-thunknallla-prd`](https://hc-thunknalla-prd.herokuapp.com)- production environment app.
  - [x] [`hc-thunknallla-dev`](https://hc-thunknalla-dev.herokuapp.com)- development app.
- [x] Configure git branches to be deployed for each of the apps:
  - [x] production is configured to deploy the [`develop`](https://github.com/andela/healthchecks-thunknalla/tree/develop) branch
  - [x] develop is configured to deploy any of the feature branches that need to be showcased

#### How should this be manually tested?
- Opening https://hc-thunknalla-prd.herokuapp.com on your favorite browser should load healthchecks-thunknalla at the state the develop branch is at.
- Opening https://hc-thunknalla-dev.herokuapp.com on your favorite browser should load healthchecks-thunknalla at the state the [branch](https://github.com/andela/healthchecks-thunknalla/tree/ch-setting_up_heroku_deployment-146143817) for this PR is at. At least until it's merged.

#### Any background context you want to provide?
- Since the [`develop`](https://github.com/andela/healthchecks-thunknalla/tree/develop) branch is the default branch for this repo, it represents the stable state of the project more than master would. Makes sense to deploy it to the production environment.
- While this PR has been initiated by @collinmutembei, the development work happened in pairing sessions to accelerate progress.

#### What are the relevant pivotal tracker stories?
The PT story for this work can be found [here](https://www.pivotaltracker.com/story/show/146143817)

#### Screenshots

prd environment
screen shot 2017-06-01 at 14 46 14
  
dev environment
screen shot 2017-06-01 at 14 46 31

### CODE REVIEW

**Comment by collinmutembei** [comment]
Travis is red @georgreen

File: hc/settings.py [hunk 1]
```diff
@@ -81,12 +83,18 @@
...
 # Default database engine is SQLite. So one can just check out code,
 # install requirements.txt and do manage.py runserver and it works
-DATABASES = {
-    'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME':   './hc.sqlite',
+if os.environ.get("PLATFORM") == "HEROKU":
+    databse_url = os.environ.get("DATABASE_URL")
+    DATABASES = {
+        'default': dj_database_url.config(default=databse_url)
```
Comment (collinmutembei) [inline hc/settings.py]: Delete [line 87](https://github.com/andela/healthchecks-thunknalla/blob/e88ed78f4a320759e980b2d2408d6ed202d30920/hc/settings.py#L87) and update this to:
```python
'default' = dj_database_url.config(conn_max_age=600)
```

File: hc/settings.py [hunk 2]
```diff
@@ -153,3 +161,7 @@
     from .local_settings import *
 else:
     warnings.warn("local_settings.py not found, using defaults")
+
+
+STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')
+STATIC_URL = '/static/'
```
Comment (collinmutembei) [inline hc/settings.py:168]: Aren't you forgetting something --> [`settings.STATICFILES_STORAGE`](https://devcenter.heroku.com/articles/django-app-configuration#whitenoise)

**Review by collinmutembei — COMMENTED** [review]
Needs work.

**Comment by collinmutembei** [comment]
Where are the screenshots?

File: hc/settings.py (hunk 2 above)
Comment (georgreen) [reply hc/settings.py:168]: This breaks Travis, I had it before and Travis failed should I add it to work only on Heroku platform?

**Review by victoriaaoka — APPROVED** [review]

**Comment by collinmutembei** [comment]
@georgreen @davidmukiibi @victoriaaoka @Andretalik @Kachulio1 
Turns out, for `whitenoise` to work, it needs to be added to `settings.MIDDLEWARE_CLASSES` as outlined [here](http://whitenoise.evans.io/en/stable/django.html#enable-whitenoise).

Who wants to rectify this?

File: hc/settings.py (hunk 2 above)
Comment (collinmutembei) [reply hc/settings.py:168]: I suppose [this](https://github.com/andela/healthchecks-thunknalla/pull/7#issuecomment-306603709) is why build [237540973](https://travis-ci.org/andela/healthchecks-thunknalla/builds/237540973) was failing.

**Pull request merged**